dto/agent_dto/case_review.go

```diff
@@ -15,12 +19,39 @@ type AiCaseReviewDto interface {
 	aiCaseReviewDto()
 }
 
+type OriginName string
+
+const (
+	OriginNameDataModel OriginName = "data_model"
+	OriginNameInternal  OriginName = "internal"
+	OriginNameUnknown   OriginName = "unknown"
+)
+
+type CaseReviewProof struct {
+	Id     string     `json:"id"`
+	Type   string     `json:"type"`
+	Origin OriginName `json:"origin"`
+	Reason string     `json:"reason"`
+}
+
+func OriginNameFromString(s string) OriginName {
+	switch s {
+	case "data_model":
+		return OriginNameDataModel
+	case "internal":
+		return OriginNameInternal
+	default:
+		return OriginNameUnknown
+	}
+}
+
 type CaseReviewV1 struct {
-	Ok          bool   `json:"ok"`
-	Output      string `json:"output"`
-	SanityCheck string `json:"sanity_check"`
-	Thought     string `json:"thought"`
-	Version     string `json:"version"`
+	Ok          bool              `json:"ok"`
+	Output      string            `json:"output"`
+	SanityCheck string            `json:"sanity_check"`
+	Thought     string            `json:"thought"`
+	Version     string            `json:"version"`
```

Comment:
Let's not serialize the version inside the review itself, since it is present one level up (unless this DTO can be sent on its own).